### ISSUE TITLE
What is rationale for two different cmbtp_* variables?

### ISSUE THREAD

**Issue by martinholmer** [open-source-economics/Tax-Calculator#1073]
In the taxdata repository, the following code (which used to be in Tax-Calculator) computes the `cmbtp_standard` and the `cmbtp_itemizer` variables:
```
    # (C) Impute cmbtp_standard and cmbtp_itemizer variables:
    data['cmbtp_standard'] = data['e62100'] - data['e00100'] + data['e00700']
    zero = np.zeros(len(data.index))
    medical_limit = np.maximum(zero, data['e17500'] -
                               np.maximum(zero, data['e00100']) * 0.075)
    med_adj = np.minimum(medical_limit,
                         0.025 * np.maximum(zero, data['e00100']))
    stx_adj = np.maximum(zero, data['e18400'])
    data['cmbtp_itemizer'] = (data['e62100'] - med_adj + data['e00700'] +
                              data['p04470'] + data['e21040'] - stx_adj -
                              data['e00100'] - data['e18500'] -
                              data['e20800'])
```
Can somebody explain the rationale for having two different variables?

This question is being posed because the recent addition of those variables into the calculation of `_expanded_income` (see pull request #1057) causes `_expanded_income` to change simply because a reform induces a filing unit to change from itemizing deductions to using the standard deduction (or vice versa).  Such a change in `_expanded_income` does not seem intuitive or sensible.

@MattHJensen @feenberg @Amy-Xu @GoFroggyRun @andersonfrailey @codykallen

**Comment by feenberg** [comment]
On Sun, 20 Nov 2016, Martin Holmer wrote:

> 
> In the taxdata repository, the following code (which used to be in Tax-Calculator)
> computes the cmbtp_standard and the cmbtp_itemizer variables:
>
>     # (C) Impute cmbtp_standard and cmbtp_itemizer variables:
>     data['cmbtp_standard'] = data['e62100'] - data['e00100'] + data['e00700']
>     zero = np.zeros(len(data.index))
>     medical_limit = np.maximum(zero, data['e17500'] -
>                                np.maximum(zero, data['e00100']) * 0.075)
>     med_adj = np.minimum(medical_limit,
>                          0.025 * np.maximum(zero, data['e00100']))
>     stx_adj = np.maximum(zero, data['e18400'])
>     data['cmbtp_itemizer'] = (data['e62100'] - med_adj + data['e00700'] +
>                               data['p04470'] + data['e21040'] - stx_adj -
>                               data['e00100'] - data['e18500'] -
>                               data['e20800'])
> 
> Can somebody explain the rationale for having two different variables?
> 
> This question is being posed because the recent addition of those variables into the
> calculation of _expanded_income (see pull request #1057) causes _expanded_income to
> change simply because a reform induces a filing unit to change from itemizing
> deductions to using the standard deduction (or vice versa). Such a change in
> _expanded_income does not seem intuitive or sensible.


Isn't the solution to this dilemma that the the choice of which definition 
to use should depend on the taxpayer decision to itemize in the base law, 
and not be affected by the reform?

dan

**Comment by martinholmer** [comment]
@feenberg said:
> Isn't the solution to this dilemma that the the choice of which definition to use should depend on the taxpayer decision to itemize in the base law, and not be affected by the reform?

Yes, I think so.  I've already started preparing pull requests in taxdata (to include the FDED variable in the `puf.csv` file) and in taxcalc (to use FDED to identify which `cmbtp_*` variable should be used to define (immediately after the data is read in and before any tax calculations) a `nonAGI_income` variable that will be added directly into `_expanded_income`.  Does that make sense?

**Comment by martinholmer** [comment]
Issue #1073 is resolved in pending pull request #1075.

**Comment by martinholmer** [comment]
Issue #1073 has been resolved by the merger of pull request #1077 into the Tax-Calculator master branch.
